torch/_functorch/aot_autograd.py

```diff
@@ -1100,12 +1227,20 @@ def inner_fn(primals: List[Any], tangents: List[Any]):
         # Call the backwards pass
         if grad_primals:
             with fx_traceback.preserve_node_meta():
-                backward_out = torch.autograd.grad(
-                    needed_outs,
-                    grad_primals,
-                    grad_outputs=needed_tangents,
-                    allow_unused=True,
-                )
+                # If our output is a scalar loss, we don't need to pass in tangents.
+                if len(needed_tangents) == 1 and needed_tangents[0].ndim == 0:
```

Comment:
loss could be 1d scalar? A more robust condition is 

`needed_tangents[0].numel() == 1`